src/libraries/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.ERR.cs

```diff
@@ -30,7 +33,13 @@ internal static partial class Crypto
 
         private static unsafe string ErrErrorStringN(ulong error)
         {
-            var buffer = new byte[1024];
+            if (t_msgBuf is null)
+            {
+                t_msgBuf = new byte[1024];
+            }
```

Comment:
Why not use ArrayPool for this?

Reply:
Probably recency bias... since I'd just had a discussion with someone about ThreadStatics :smile:  Switching to the pool, there's no reason to hold on to the 1k/throwing-thread.